third_party/2and3/boto/utils.pyi

```diff
@@ -0,0 +1,215 @@
+    sep: Optional[str] = ...,
+    url: str = ...,
+    timeout: Optional[int] = ...,
+    num_retries: int = ...,
+) -> Dict[str, str]: ...
+
+ISO8601 = ...  # type: str
+ISO8601_MS = ...  # type: str
+RFC1123 = ...  # type: str
+LOCALE_LOCK = ...  # type: _LockType
+
+def setlocale(name: Union[str, Tuple[str, str]]) -> None: ...
+def get_ts(ts: Optional[time.struct_time] = ...) -> str: ...
+def parse_ts(ts: str) -> datetime.datetime: ...
+def find_class(module_name: str, class_name: Optional[str] = ...) -> Optional[Type[Any]]: ...
+def update_dme(username: str, password: str, dme_id: str, ip_address: str) -> str: ...
+def fetch_file(
+    uri: str,
+    file: Optional[IO] = ...,
+    username: Optional[str] = ...,
+    password: Optional[str] = ...,
+) -> IO: ...
+
+class ShellCommand:
+    exit_code = ...  # type: int
+    command = ...  # type: subprocess._CMD
+    log_fp = ...  # type: IO
+    wait = ...  # type: bool
+    fail_fast = ...  # type: bool
+
+    def __init__(
+        self,
+        command: subprocess._CMD,
+        wait: bool = ...,
+        fail_fast: bool = ...,
+        cwd: Optional[subprocess._TXT] = ...,
+    ) -> None: ...
+
+    process = ...  # type: subprocess.Popen
+
+    def run(self, cwd: Optional[subprocess._CMD] = ...) -> Optional[int]: ...
+    def setReadOnly(self, value) -> None: ...
+    def getStatus(self) -> Optional[int]: ...
+
+    status = ...  # type: Optional[int]
+
+    def getOutput(self) -> str: ...
+
+    output = ...  # type: str
+
+class AuthSMTPHandler(logging.handlers.SMTPHandler):
+    username = ...  # type: str
+    password = ...  # type: str
+    def __init__(
+        self,
+        mailhost: str,
+        username: str,
+        password: str,
+        fromaddr: str,
+        toaddrs: Sequence[str],
+        subject: str,
+    ) -> None: ...
+    def emit(self, record: logging.LogRecord) -> None: ...
+
+class LRUCache(Dict[_KT, _VT]):
+    class _Item:
+        previous = ...  # type: Optional[LRUCache._Item]
+        key = ...
+        value = ...
+        def __init__(self, key, value) -> None: ...
+
+    capacity = ...  # type: int
+    head = ...  # type: Optional[LRUCache._Item]
+    tail = ...  # type: Optional[LRUCache._Item]
+
+    def __init__(self, capacity: int) -> None: ...
+
+
+# This exists to work around Password.str's name shadowing the str type
+_str = str
+
+class Password:
+    hashfunc = ...  # type: Callable
+    str = ...  # type: Optional[_str]
+
+    def __init__(
+        self,
+        str: Optional[_str] = ...,
+        hashfunc: Optional[Callable] = ...,
+    ) -> None: ...
+    def set(self, value: Union[bytes, _str]) -> None: ...
+    def __str__(self) -> _str: ...
+    def __eq__(self, other: Any) -> bool: ...
+    def __len__(self) -> int: ...
+
+def notify(
+    subject: str,
+    body: Optional[str] = ...,
+    html_body: Optional[Union[Sequence[str], str]] = ...,
+    to_string: Optional[str] = ...,
+    attachments: Optional[Iterable] = ...,
```

Comment:
What kind of `Iterable`?

Reply:
Each attachment is passed to `email.message.Message.attached` which is typed to be a `Message` https://github.com/python/typeshed/blob/7ecc9792112d5e60801a4aa9af05a32002b8c930/stdlib/3/email/message.pyi#L31